pkg/querier/queryrange/roundtrip.go

```diff
@@ -46,12 +47,23 @@ type Config struct {
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxRetries, "querier.max-retries-per-request", 5, "Maximum number of retries for a single request; beyond this, the downstream error is returned.")
 	f.BoolVar(&cfg.SplitQueriesByDay, "querier.split-queries-by-day", false, "Deprecated: Split queries by day and execute in parallel.")
-	f.DurationVar(&cfg.SplitQueriesByInterval, "querier.split-queries-by-interval", 0, "Split queries by an interval and execute in parallel, 0 disables it. You should use an a multiple of 24 hours (same as the storage bucketing scheme), to avoid queriers downloading and processing the same chunks.")
+	f.DurationVar(&cfg.SplitQueriesByInterval, "querier.split-queries-by-interval", 0, "Split queries by an interval and execute in parallel, 0 disables it. You should use an a multiple of 24 hours (same as the storage bucketing scheme), to avoid queriers downloading and processing the same chunks. This also determines how cache keys are chosen when result caching is enabled")
 	f.BoolVar(&cfg.AlignQueriesWithStep, "querier.align-querier-with-step", false, "Mutate incoming queries to align their start and end with their step.")
 	f.BoolVar(&cfg.CacheResults, "querier.cache-results", false, "Cache query results.")
 	cfg.ResultsCacheConfig.RegisterFlags(f)
 }
 
+func (cfg *Config) Validate() error {
+	if cfg.SplitQueriesByDay {
+		cfg.SplitQueriesByInterval = day
```

Comment:
Can you add a log warning if `cfg.SplitQueriesByInterval` is not set to the default value? You could also suggest in the message to stop using `SplitQueriesByDay` in favor of setting `SplitQueriesByInterval` to `24h` since we are deprecating it.

Reply:
This warning was already logged during initialization, I've just moved the override here. To address your feedback, I've now added a logger as an argument to our validation functions. One consequence is we're now initializing the logger based on the server config before the configs are validated [here](https://github.com/cortexproject/cortex/pull/2040/files#diff-cf2ba502f0bdf282c97f46997569748cR76-R79). It looks reasonable as the log levels are set with defaults, but let me know if I should revert/do something else.